src/core/memory.cpp

```diff
@@ -238,13 +251,31 @@ int MemoryManager::MapFile(void** out_addr, VAddr virtual_addr, size_t size, Mem
 void MemoryManager::UnmapMemory(VAddr virtual_addr, size_t size) {
     std::scoped_lock lk{mutex};
 
-    // TODO: Partial unmaps are technically supported by the guest.
-    const auto it = vma_map.find(virtual_addr);
-    ASSERT_MSG(it != vma_map.end() && it->first == virtual_addr,
-               "Attempting to unmap partially mapped range");
-
-    const auto type = it->second.type;
+    auto vma_handle = FindVMA(virtual_addr);
+    auto& entire_vma = vma_handle->second;
+    const VAddr start_in_vma = virtual_addr - entire_vma.base;
+    const VAddr end_in_vma = start_in_vma + size;
+    const auto type = entire_vma.type;
     const bool has_backing = type == VMAType::Direct || type == VMAType::File;
+
+    // There does not appear to be comparable support for partial unmapping on Windows.
```

Comment:
The memory system should not handle OS specific operations. Move this logic inside address space

Reply:
You can probably use the placeholders array to check which region is currently mapped

Reply:
Alright that makes sense. I'll move the logic into Unmap().